### PR TITLE
DOCS-5291 React Native docs improvements

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What does this PR do?

Add a few improvements to the RN docs before un-single sourcing.
Also ports the following PRs that were not merged:
- https://github.com/DataDog/dd-sdk-reactnative/pull/456
- https://github.com/DataDog/dd-sdk-reactnative/pull/458

### Reviewer checklist
- [x] Review the changed files.
- [x] Review the URLs listed in the [Preview](#preview) section.
- [x] Check images for PII
- [x] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/real_user_monitoring/reactnative/_index.md [hunk 1]
```diff
@@ -285,6 +291,8 @@ Use one of Datadog's integrations to automatically track views for the following
 -   If you use the [`react-native-navigation`][5] library, then add the `@datadog/mobile-react-native-navigation` package and follow the [setup instructions][6].
 -   If you use the [`react-navigation`][7] library, then add the `@datadog/mobile-react-navigation` package and follow the [setup instructions][8].
 
+If you experience any issue setting up View tracking with `@datadog/mobile-react-navigation` you can check out our [example application][16].
```
Comment (rtrieu) [inline content/en/real_user_monitoring/reactnative/_index.md:294]:
```suggestion
If you experience any issues setting up View tracking with `@datadog/mobile-react-navigation` you can see our [example application][16] as a reference.
```

File: content/en/real_user_monitoring/reactnative/codepush.md [hunk 2]
```diff
@@ -20,6 +20,8 @@ Each time you release a new [CodePush][1] version for your React Native applicat
 
 Datadog recommends using `@datadog/mobile-react-native-code-push` in your app and the [datadog-ci][3] `react-native codepush` command to upload your source maps. This ensures that the `version` is consistent in both reported crashes and uploaded source maps.
 
+If you experience any issue setting up the Datadog SDK with codepush you can check out our [example application][6].
```
Comment (rtrieu) [inline content/en/real_user_monitoring/reactnative/codepush.md:23]:
```suggestion
If you experience any issues setting up the Datadog SDK with codepush, you can see our [example application][6] as a reference.
```

File: content/en/real_user_monitoring/reactnative/expo.md [hunk 3]
```diff
@@ -17,6 +17,8 @@ The RUM React Native SDK supports Expo and Expo Go. To use it, install `expo-dat
 
 `expo-datadog` supports Expo starting from SDK 45 and the plugin's versions follow Expo versions. For example, if you use Expo SDK 45, use `expo-datadog` version `45.x.x`. Datadog recommends using **Expo SDK 45** as a minimum version; previous versions may require manual steps.
 
+If you experience any issue setting up the datadog SDK with an expo application you can check out our [example application][8].
```
Comment (rtrieu) [inline content/en/real_user_monitoring/reactnative/expo.md:20]:
```suggestion
If you experience any issues setting up the Datadog SDK with an Expo application, you can see our [example application][8] as a reference.
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
A few minor suggestions, but otherwise looks good!

File: content/en/real_user_monitoring/reactnative/_index.md [hunk 4]
```diff
@@ -380,10 +388,6 @@ end
 
 **Note:** This solution comes from this [StackOverflow][14] post.
```
Comment (rtrieu) [inline content/en/real_user_monitoring/reactnative/_index.md:389]:
```suggestion
**Note**: This solution comes from this [StackOverflow][14] post.
```

File: content/en/real_user_monitoring/reactnative/advanced_configuration.md [hunk 5]
```diff
@@ -215,6 +215,8 @@ All interactions with the RUM SDK (view tracking, actions, resources tracing, an
 
 Logs are not recorded and calling a `DdLogs` method before the actual initialization might break logging.
 
+If you experience any issue setting up the asynchronous initialization of Datadog you can check out our [example application][7].
```
Comment (rtrieu) [inline content/en/real_user_monitoring/reactnative/advanced_configuration.md:218]:
```suggestion
If you experience any issue setting up the asynchronous initialization of Datadog, you can check out our [example application][7].
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
A few very minor suggestions, but otherwise looks good!

**Review by rtrieu — APPROVED** [review]
Looks good, thanks!

**Comment by louiszawadzki** [comment]
I don't have the rights to merge, I'll let you do it :)

**Review by xgouchet — APPROVED** [review]

**Pull request merged**